src/google-maps/map-marker-clusterer/map-marker-clusterer.ts

```diff
@@ -0,0 +1,603 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="googlemaps" />
+
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+  QueryList,
+  ViewEncapsulation
+} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, Subject} from 'rxjs';
+import {map, take, takeUntil} from 'rxjs/operators';
+
+import {GoogleMap} from '../google-map/google-map';
+import {MapEventManager} from '../map-event-manager';
+import {MapMarker} from '../map-marker/map-marker';
+
+/**
+ * Angular component for implementing a Google Maps Marker Clusterer.
+ *
+ * See https://developers.google.com/maps/documentation/javascript/marker-clustering
+ */
+@Component({
+  selector: 'map-marker-clusterer',
+  exportAs: 'mapMarkerClusterer',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: '<ng-content></ng-content>',
+  encapsulation: ViewEncapsulation.None,
+})
+export class MapMarkerClusterer implements OnInit, AfterContentInit, OnDestroy {
+  private readonly _ariaLabelFn = new BehaviorSubject<AriaLabelFn|undefined>(undefined);
+  private readonly _averageCenter = new BehaviorSubject<boolean|undefined>(undefined);
+  private readonly _batchSizeIE = new BehaviorSubject<number|undefined>(undefined);
+  private readonly _calculator = new BehaviorSubject<Calculator|undefined>(undefined);
+  private readonly _clusterClass = new BehaviorSubject<string|undefined>(undefined);
+  private readonly _enableRetinalIcons = new BehaviorSubject<boolean|undefined>(undefined);
+  private readonly _gridSize = new BehaviorSubject<number|undefined>(undefined);
+  private readonly _ignoreHidden = new BehaviorSubject<boolean|undefined>(undefined);
+  private readonly _imageExtension = new BehaviorSubject<string|undefined>(undefined);
+  private readonly _imagePath = new BehaviorSubject<string|undefined>(undefined);
+  private readonly _imageSizes = new BehaviorSubject<number[]|undefined>(undefined);
+  private readonly _maxZoom = new BehaviorSubject<number|undefined>(undefined);
+  private readonly _minimumClusterSize = new BehaviorSubject<number|undefined>(undefined);
+  private readonly _styles = new BehaviorSubject<ClusterIconStyle[]|undefined>(undefined);
+  private readonly _title = new BehaviorSubject<string|undefined>(undefined);
+  private readonly _zIndex = new BehaviorSubject<number|undefined>(undefined);
+  private readonly _zoomOnClick = new BehaviorSubject<boolean|undefined>(undefined);
```

Comment:
More of a general observation: I don't think that our current approach with a subject and a setter for each property is very scalable. What do you think about reworking it? I can change one component to what I think we should do and we can discuss it on the PR?

Reply:
I would definitely consider that (I assume something with ngOnChanges?) so that sounds good. I'd prefer to do that kind of refactoring in a different pr though.

Reply:
Yeah, I was imagining going through `ngOnChanges` and turning everything into regular properties. Agreed that it shouldn't be in this PR or all the components at the same time. I'll change only one component so we can agree on the approach and we can take it from there.